src/Repository/UserRepository.php

```diff
@@ -40,6 +40,17 @@ public function upgradePassword(PasswordAuthenticatedUserInterface $user, string
         $this->getEntityManager()->flush();
     }
 
+    public function findUserByEmail($email): array
+    {
+        return $this->createQueryBuilder('u')
+            ->andWhere('u.email = :val')
+            ->setParameter('val', $email)
+            ->orderBy('u.id', 'ASC')
+            ->setMaxResults(1)
```

Comment:
Your variable should be easy to understand. 
`u` doesn't mean anything.

Reply:
I fix that named "u" for "user"